test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -4026,6 +4027,63 @@ public async Task TaskHubName_DefaultNameNonProductionSlot_ThrowsException()
             }
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task TaskHubName_AppSettingReference_ValidTaskHub_UsesResolvedTaskHub()
+        {
+            string taskHubSettingName = "TaskHubName";
+            string taskHubName = "ValidTaskHub";
+            DurableTaskOptions durableTaskOptions = new DurableTaskOptions();
+            durableTaskOptions.HubName = $"%{taskHubSettingName}%";
+
+            var mockNameResolver = new Mock<INameResolver>();
+            mockNameResolver.Setup(res => res.Resolve(taskHubSettingName))
+                .Returns(taskHubName);

```

Comment:
Instead of mocking, it might be more straightforward to use `SimpleNameResolver`.